slides.Rmd

```diff
@@ -54,53 +56,107 @@ table_contents2 <- data.frame(c("&", "|",
+## Difference between R and RStudio?
 
+- RStudio is a free and open-source integrated development environment for R, a programming language for statistical computing and graphics. You cannot run RStudio without R.
 - RStudio is more flexible and powerful, and provides direct access to R code
 - Both are good for viewing data, neither is good for manually editing data
 - Use RStudio for any project that requires direct interaction with code and/or manipulation of complex data
+- RStudio Pro is proprietary and comes with user support from RStudio and some extra functionality like the use of RStudio server.
 
-## Functions and data embedded in R
 
-```{r,  eval=FALSE, echo=TRUE}
-summary(cars)
+## RStudio
+![RStudio](images/RStudio_Screenshot.png){ width=100%, height=100%}
+```{r, echo = FALSE}
+# TRAINERS NOTES
+# Explain what each window does
+# Note that the console is for instantly executing a line of code and is different from the script
 ```
 
-```{r}
-summary(cars)
+
+## R packages 
+
+- R packages are collections of functions and, sometimes, datasets created by the R community. 
+
+- R packages cover a wide range of needs and can improve the basic functionality of R or add additional functionality. 
+
+- Packages come with extensive helpfiles called vignettes. These contain information on all the functions in the package, a description of what the packaage is designed to do
+
+- Link to package vignette - suggestions of a good one?
+
+## Installing packages
+
+- Packages are installed using:
+```{r, echo = TRUE, eval = FALSE}
+install.packages("haven")
 ```
 
-## Easy to plot charts 
+- Packages are called using:
+```{r, echo = TRUE, results = "hide"}
+library("haven")
+```
+ 
+```{r, echo = FALSE}
+# TRAINERS NOTES
+# Explain the difference between installing and calling, a package is not a library)
+```
 
-```{r,  eval=FALSE, echo=TRUE}
-plot(cars)
+- As anyone can create an R package, different packages can have functions of the same name. 
+
+```{r, echo = FALSE}
+# TRAINERS NOTES
+# If you have loaded two packages that have a function of the same name, R uses the one from the package loaded last.
 ```
 
-```{r}
-plot(cars)
+- We can specify which package a function should be from using ::
+```{r, echo = TRUE, eval = FALSE}
+dplyr::summarise()
 ```
 
 
-## R packages 
+## Data Frames and Objects
+**Data Frame** <br/>
+- A data frame is a table of information in which each column contains values of one variable and each row contains all the information on a single observation. 
 
-Packages allow you to use R much more efficiently. Rather than simply relying on some basic R techniques, they build on some widely used R methods and often make complex tasks very straight forward. R packages cover a wide range of needs. 
+**Objects**<br/>
+- can't remember what we were going to put here? Any suggestions welcome!
 
-Try typing:
 
+## Comparison Operators
```

Comment:
I don't think that comparison and logical operators fit here. I get that its basic stuff but it will be a while before they are actually used (especially with long breaks between sessions).

Reply:
I agree, I think they might fit better in block 3 for data manipulation.

Reply:
Thanks both, have moved this part.